internal/pkg/template/input/input.go

```diff
@@ -0,0 +1,57 @@
+package input
+
+import (
+	"context"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/validator"
+)
+
+type Inputs []*Input
+
+func (i Inputs) Validate() error {
+	validations := []validator.Validation{
+		{
+			Tag:  "template-input-id",
+			Func: validateInputId,
+		},
+		{
+			Tag:  "template-input-default",
+			Func: validateInputDefault,
+		},
+		{
+			Tag:  "template-input-options",
+			Func: validateInputOptions,
+		},
+		{
+			Tag:  "template-input-type",
+			Func: validateInputType,
+		},
+		{
+			Tag:  "template-input-rules",
+			Func: validateInputRules,
+		},
+	}
+	return validator.Validate(i, validations...)
+}
+
+type Input struct {
+	Id          string      `json:"id" validate:"required,template-input-id"`
+	Name        string      `json:"name" validate:"required"`
+	Description string      `json:"description" validate:"required"`
+	Default     interface{} `json:"default,omitempty" validate:"omitempty,template-input-default"`
+	Kind        string      `json:"kind" validate:"required,oneof=input password textarea confirm select multiselect"`
+	Type        string      `json:"type,omitempty" validate:"required_if=Kind input,omitempty,oneof=string int float64,template-input-type"`
+	Options     []Option    `json:"options,omitempty" validate:"required_if=Type select Type multiselect,template-input-options"`
+	Rules       string      `json:"rules,omitempty" validate:"template-input-rules"`
+	If          string      `json:"if,omitempty"`
+}
+
+func (i Input) Validate(userInput interface{}, ctx context.Context) error {
```

Comment:
A tuto metodu `ValidateUserInput`? 
V kode by sa to pouzivalo `input.ValidateUserInput(value, ctx)` 

+ pridal by som komentar: `... validuje vstup zadany od uzivatela pomocou Input.Rules`